nano/nano_node/daemon.cpp

```diff
@@ -19,8 +20,69 @@
 #include <boost/process.hpp>
 #endif
 
+// Some builds (mac) fail due to "Boost.Stacktrace requires `_Unwind_Backtrace` function".
+#ifndef _WIN32
+#ifndef _GNU_SOURCE
+#define BEFORE_GNU_SOURCE 0
+#define _GNU_SOURCE
+#else
+#define BEFORE_GNU_SOURCE 1
+#endif
+#endif
+// On Windows this include defines min/max macros, so keep below other includes
+// to reduce conflicts with other std functions
+#include <boost/stacktrace.hpp>
+#ifndef _WIN32
+#if !BEFORE_GNU_SOURCE
+#undef _GNU_SOURCE
+#endif
+#endif
+
+namespace
+{
+#ifdef __linux__
+#include <link.h>
+// Only on linux. This outputs the load addresses for the executable and shared libraries.
+// Useful for debugging should the virtual addresses be randomized.
+int output_memory_load_address (dl_phdr_info * info, size_t, void *)
+{
+	static int counter = 0;
+	std::ostringstream ss;
+	ss << "nano_node_crash_load_address_dump_" << counter << ".txt";
+	std::ofstream file (ss.str ());
+	file << "Name: " << info->dlpi_name << "\n";
+
+	for (auto i = 0; i < info->dlpi_phnum; ++i)
+	{
+		// Only care about the first load address
+		if (info->dlpi_phdr[i].p_type == PT_LOAD)
+		{
+			file << std::hex << (void *)(info->dlpi_addr + info->dlpi_phdr[i].p_vaddr);
+			break;
+		}
+	}
+	++counter;
+	return 0;
+}
+#endif
+
+void my_abort_signal_handler (int signum)
+{
+	::signal (signum, SIG_DFL);
```

Comment:
Will this generate a core dump if enabled? Some info I've read indicates a call to kill with the same signal is required in addition to calling with SIG_DFL.

Reply:
Using the boost stacktrace example it calls `::signal (signum, SIG_ABRT)` at the end, which didn't generate a core-dump (never said it would, just something I noticed). I also saw some people using `kill` at the end as well, but again saw no core dump 👎 , letting it fall through seemed to work though. Would be good if others could test this as well for greater coverage.